tests/unit/misc/test_autoupdate.py

```diff
@@ -0,0 +1,104 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.misc.autoupdate"""
+
+from PyQt5.QtTest import QSignalSpy
+
+from qutebrowser.misc import autoupdate, httpclient
+
+
+class HTTPGetStub(httpclient.HTTPClient):
+    """A stub class for HTTPClient.
+
+        Attributes:
+        _success: Wether get() will emit a success signal.
+    """
+
+    def __init__(self, success=True, json=None):
+        self._success = success
+        if json:
+            self._json = json
+        else:
+            self._json = '{"info": {"version": "test"}}'
+
+        super(HTTPGetStub, self).__init__()
+
+    def get(self, url):
+        if self._success is True:
+            self.success.emit(self._json)
+        else:
+            self.error.emit("error")
+
+
+def test_constructor():
+    client = autoupdate.PyPIVersionClient()
+    assert isinstance(client._client, httpclient.HTTPClient)
+
+
+def test_get_version_success(qtbot):
+    """Test get_version() when success is emitted."""
+    http_stub = HTTPGetStub(success=True)
+    client = autoupdate.PyPIVersionClient(client=http_stub)
+
+    # Use a spy to inspect the signal
+    success_spy = QSignalSpy(client.success)
+    error_spy = QSignalSpy(client.error)
+
+    with qtbot.waitSignal(client.success, timeout=2000, raising=False):
```

Comment:
You could drop the `success_spy` and use `raising=True` instead. I also think you can leave the timeout off, the default of 1s should be fine.